server/src/main/java/org/jetbrains/bsp/bazel/server/sync/languages/rust/RustWorkspaceResolver.kt

```diff
@@ -0,0 +1,278 @@
+package org.jetbrains.bsp.bazel.server.sync.languages.rust
+
+import ch.epfl.scala.bsp4j.RustDepKindInfo
+import ch.epfl.scala.bsp4j.RustDependency
+import ch.epfl.scala.bsp4j.RustEnvData
+import ch.epfl.scala.bsp4j.RustFeature
+import ch.epfl.scala.bsp4j.RustPackage
+import ch.epfl.scala.bsp4j.RustProcMacroArtifact
+import ch.epfl.scala.bsp4j.RustRawDependency
+import ch.epfl.scala.bsp4j.RustTarget
+import org.jetbrains.bsp.bazel.server.sync.BazelPathsResolver
+import org.jetbrains.bsp.bazel.server.sync.model.Label
+import org.jetbrains.bsp.bazel.server.sync.model.Module
+
+private data class BazelPackageTargetInfo(
+    val packageName: String,
+    val targetName: String
+)
+
+private data class RustTargetModule(
+    val module: Module,
+    val rustModule: RustModule
+)
+
+class RustWorkspaceResolver(val bazelPathsResolver: BazelPathsResolver) {
```

Comment:
quite a lot of things is still happening here + the order of the methods should be changed - https://kotlinlang.org/docs/coding-conventions.html#class-layout the declaration of a method should be below it's usage, and maybe it's possible to split it even more into smaller classes

Reply:
I split `RustWorkspaceResolver` into `RustPackageResolver` and `RustDependencyResolver` so it should be clearer now